internal/impl/mongodb/client/config.go

```diff
@@ -121,5 +232,21 @@ func ConfigDocs() docs.FieldSpecs {
 		docs.FieldString("database", "The name of the target MongoDB DB."),
 		docs.FieldString("username", "The username to connect to the database."),
 		docs.FieldString("password", "The password to connect to the database."),
+		docs.FieldBool("password_set", "For GSSAPI, this must be true if a password is specified, even if the password is the empty string, and false if no password is specified, indicating that the password should be taken from the context of the running process. For other mechanisms, this field is ignored.").Optional().Advanced().AtVersion("4.11.0"),
+		docs.FieldString("auth_source", `The name of the database to use for authentication. This can also be set through the "authSource" URI option (e.g. "authSource=otherDb"). For more information, see [MongoDB docs](https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.authSource).`).Optional().Advanced().AtVersion("4.11.0"),
+		docs.FieldString("auth_mechanism", `The mechanism to use for authentication. This can also be set through the "authMechanism" URI option. (e.g. "authMechanism=PLAIN"). For more information, see [MongoDB docs](https://docs.mongodb.com/manual/core/authentication-mechanisms/).`).Optional().Advanced().AtVersion("4.11.0"),
+		docs.FieldObject("auth_mechanism_properties", `Used to specify additional configuration options for certain mechanisms. For more information, see [MongoDB docs](https://docs.mongodb.com/manual/core/authentication-mechanisms/).`).Optional().Advanced().AtVersion("4.11.0").WithChildren(
+			docs.FieldString("service_name", "The service name to use for GSSAPI authentication.").Optional(),
+			docs.FieldBool("canonicalize_host_name", "Whether to canonicalize the host name for GSSAPI authentication.").Optional(),
+			docs.FieldString("service_realm", "The service realm to use for GSSAPI authentication.").Optional(),
+			docs.FieldString("service_host", "The service host to use for GSSAPI authentication. The service_host and canonicalize_host_name properties must not be used at the same time on Linux and Darwin systems.").Optional(),
```

Comment:
> The service_host and canonicalize_host_name properties must not be used at the same time on Linux and Darwin systems.

Interesting! Do they have to be used at the same time on Windows? I think we could add a lint rule for this where `runtime.GOOS` is concatenated into the bloblang code.

Reply:
Hey @mihaitodor how can I do a lint rule in this API? I saw a LinterFunc but I don't know how to access other fields in it..

Reply:
There are a few examples [here](https://github.com/benthosdev/benthos/blob/3310065cf97944a8545563187997cc47e409fa92/internal/impl/snowflake/output_snowflake_put.go#L197-L209). Not sure if that helps you.